torchreid/engine/image/am_softmax.py

```diff
@@ -377,4 +412,175 @@ def _apply_batch_transform(self, imgs, obj_ids):
             permuted_idx = torch.randperm(imgs.shape[0])
             imgs = lambd * imgs + (1 - lambd) * imgs[permuted_idx]
 
+        if self.aug_type == 'fmix':
+            r = np.random.rand(1)
+            if self.alpha > 0 and r[0] <= self.aug_prob:
+                lam, fmask = sample_mask(self.alpha, self.decay_power, self.size,
+                                        self.max_soft, self.reformulate)
+                index = torch.randperm(imgs.size(0)).to(imgs.device)
```

Comment:
Exclude the allocation on CPU if you want to use GPU next: `index = torch.randperm(imgs.size(0), device=imgs.device)`